packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2465,4 +2600,18 @@ export class SnapController extends BaseController<
 
     return { newPermissions, unusedPermissions, approvedPermissions };
   }
+
+  #isValidUpdate(snapId: SnapId, newVersionRange: SemVerRange) {
+    const existingSnap = this.getExpect(snapId);
+
+    if (satisfiesVersionRange(existingSnap.version, newVersionRange)) {
+      return false;
+    }
```

Comment:
I'm not sure if I understand the logic here. Shouldn't this return `true`?

Reply:
This function is trying to gauge whether we are in need of an update. If the existing snap satisfies the version range, we don't need to update.

Reply:
No, because we don't want to create a snapshot in the case where we are going to simply return an existing snap.

Reply:
Please explain that in the JSDoc!